### PR TITLE
Eld King Khaled chef

### PR DESCRIPTION
**Pull request by elaeon**

### CODE REVIEW

File: sushichef.py [hunk 1]
```diff
@@ -0,0 +1,477 @@
...
+    while tries < 4:
+        try:
+            document = downloader.read(source_id, loadjs=False, session=sess)
+        except requests.exceptions.HTTPError as e:
+            LOGGER.info("Error: {}".format(e))
+        except requests.exceptions.ConnectionError:
+            ### this is a weird error, may be it's raised when the webpage
+            ### is slow to respond requested resources
+            LOGGER.info("Connection error, the resource will be scraped in 5s...")
+            time.sleep(3)
```
Comment (kollivier) [inline sushichef.py]: Is it 5 seconds, or 3? :)

File: sushichef.py [hunk 2]
```diff
@@ -0,0 +1,477 @@
...
+        except requests.exceptions.TooManyRedirects as e:
+            LOGGER.info("Error: {}".format(e))
+        else:
+            return document
+        tries += 1
+    return False
+
+
+def clean_leafs_nodes_plus(channel_tree):
```
Comment (kollivier) [inline sushichef.py]: I see this method often pops up in your chefs. Is it the same code in each chef? If so, if you write some documentation and submit a PR for ricecooker, I'll be happy to add it.

Comment (kollivier) [reply sushichef.py]: Or if this and other methods could end up in some SushiChef subclass, we could add it in that way as well.

File: utils.py [hunk 3]
```diff
@@ -0,0 +1,126 @@
...
+
+
+def get_name_from_url(url):
+    head, tail = ntpath.split(url)
+    params_index = tail.find("&")
+    if params_index != -1:
+        tail = tail[:params_index]
+    params_index = tail.find("?")
+    if params_index != -1:
+        tail = tail[:params_index]
```
Comment (kollivier) [inline utils.py:30]: Why not just check for ? and trim everything after that, then just run basename on the resulting url? Also, I'm guessing this doesn't need to handle absolute URLs with protocols?

**Review by kollivier — APPROVED** [review]

File: sushichef.py (hunk 1 above)
Comment (elaeon) [reply sushichef.py]: You are right! and I removed this method from the chef because is not longer used.

File: utils.py (hunk 3 above)
Comment (elaeon) [reply utils.py:30]: I don't exactly remember why but I think I added this case to cover bad former urls or when '?' is not present in an url.

**Pull request merged**